src/components/NavGroup/NavGroup.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+
+export const NavGroup: React.FC = ({ children }) => {
+  const mapped = React.Children.map(children, (child, index) => {
+    if (React.isValidElement(child)) {
+      return React.cloneElement(child, {
+        ...child.props,
+        isFirst: index === 0,
+        isLast: !Array.isArray(children) || index === children.length - 1,
+      });
+    }
+    return null;
+  });
```

Comment:
filter out invalid elements first, otherwise `children.length` can't be trusted for `isLast` & `isFirst` checks when some components are enabled / disabled conditionally.

Reply:
fixed